src/en/cutiecomics/src/eu/kanade/tachiyomi/extension/en/cutiecomics/CutieComics.kt

```diff
@@ -0,0 +1,150 @@
+    override fun latestUpdatesRequest(page: Int): Request {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesSelector(): String {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesFromElement(element: Element): SManga {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesNextPageSelector(): String? {
+        throw UnsupportedOperationException()
+    }
+
+    // =============================== Search ===============================
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        return if (query.startsWith(PREFIX_SEARCH)) { // URL intent handler
+            val id = query.removePrefix(PREFIX_SEARCH)
+            client.newCall(GET("$baseUrl/$id"))
+                .asObservableSuccess()
+                .map(::searchMangaByIdParse)
+        } else {
+            super.fetchSearchManga(page, query, filters)
+        }
+    }
+
+    private fun searchMangaByIdParse(response: Response): MangasPage {
+        val doc = response.asJsoup()
+        val details = mangaDetailsParse(doc)
+            .apply { setUrlWithoutDomain(doc.location()) }
+        return MangasPage(listOf(details), false)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        require(query.isNotBlank() && query.length >= 4) { "Invalid search! It should have at least 4 non-blank characters." }
+        val body = FormBody.Builder()
+            .add("do", "search")
+            .add("subaction", "search")
+            .add("full_search", "0")
+            .add("search_start", "$page")
+            .add("result_from", "${(page - 1) * 20 + 1}")
+            .add("story", query)
+            .build()
+        return POST("$baseUrl/index.php?do=search", headers, body)
+    }
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    // =========================== Manga Details ============================
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        status = SManga.COMPLETED
+        update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+
+        title = document.selectFirst("h1#page-title")!!.text()
+        thumbnail_url = document.selectFirst("div.galery > img")?.absUrl("src")
+        genre = document.select("h3.field-label ~ span").joinToString { it.text() }
+    }
+
+    // ============================== Chapters ==============================
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val chapter = SChapter.create().apply {
+            url = manga.url
+            chapter_number = 1F
+            name = "Hentai"
```

Comment:
Usually the placeholder chapter name for image gallery-type sites is "Chapter"

Reply:
That or "Gallery"